lib/web_ui/dev/driver_manager.dart

```diff
@@ -60,6 +62,61 @@ class ChromeDriverManager extends DriverManager {
   }
 }
 
+/// [DriverManager] implementation for Firefox.
+///
+/// This manager can be used for both MacOS and Linux.
+class FirefoxDriverManager extends DriverManager {
+  FirefoxDriverManager(String browser) : super(browser);
+
+  FirefoxDriverInstaller firefoxDriverInstaller =
+      FirefoxDriverInstaller(geckoDriverVersion: getLockedDeckoDriverVersion());
+
+  Future<void> _installDriver() async {
+    if (_browserDriverDir.existsSync()) {
+      _browserDriverDir.deleteSync(recursive: true);
+    }
+
+    _browserDriverDir.createSync(recursive: true);
+    temporaryDirectories.add(_drivers);
+
+    final io.Directory temp = io.Directory.current;
+    io.Directory.current = _browserDriverDir;
+
+    try {
+      await firefoxDriverInstaller.install(alwaysInstall: false);
+    } finally {
+      io.Directory.current = temp;
+    }
+  }
+
+  /// Throw an error if driver directory does not exists.
+  ///
+  /// Driver should already exist on LUCI as a CIPD package.
+  Future<void> _verifyDriverForLUCI() {
+    if (!_browserDriverDir.existsSync()) {
+      throw StateError('Failed to locate Firefox driver on LUCI on path:'
+          '${_browserDriverDir.path}');
+    }
+    return Future<void>.value();
```

Comment:
Why return asynchronously?

Reply:
This method is overriding the abstract class' method. I added `@override` everywhere to make it more clear.

In this class we always return `return Future<void>.value();` value. For example line 143.